Using MAPPING_STR=00022211, VERSION_DIR=v4_enggpt
Running /platform/source/s01/v4_enggpt/m00022211/m00022211_q10_t0.sh
/platform/source/s01/v4_enggpt/m00022211/m00022211_q10_t0.sh: line 4: sudo: command not found
/platform/source/s01/v4_enggpt/m00022211/m00022211_q10_t0.sh: line 5: sudo: command not found
Requirement already satisfied: pymysql in /usr/local/lib/python3.8/dist-packages (1.1.0)
Requirement already satisfied: pymongo in /usr/local/lib/python3.8/dist-packages (4.6.0)
Requirement already satisfied: pandas in /usr/local/lib/python3.8/dist-packages (2.0.3)
Requirement already satisfied: direct-redis in /usr/local/lib/python3.8/dist-packages (0.3.1)
Requirement already satisfied: dnspython<3.0.0,>=1.16.0 in /usr/local/lib/python3.8/dist-packages (from pymongo) (2.4.2)
Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.8/dist-packages (from pandas) (2023.3.post1)
Requirement already satisfied: python-dateutil>=2.8.2 in /usr/local/lib/python3.8/dist-packages (from pandas) (2.8.2)
Requirement already satisfied: numpy>=1.20.3; python_version < "3.10" in /usr/local/lib/python3.8/dist-packages (from pandas) (1.24.4)
Requirement already satisfied: tzdata>=2022.1 in /usr/local/lib/python3.8/dist-packages (from pandas) (2023.3)
Requirement already satisfied: redis==3.4.1 in /usr/local/lib/python3.8/dist-packages (from direct-redis) (3.4.1)
Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.8/dist-packages (from python-dateutil>=2.8.2->pandas) (1.16.0)
Running /platform/source/s01/v4_enggpt/m00022211/m00022211_q11_t0.sh
Requirement already satisfied: pymysql in /usr/local/lib/python3.8/dist-packages (1.1.0)
Requirement already satisfied: pandas in /usr/local/lib/python3.8/dist-packages (2.0.3)
Requirement already satisfied: redis in /usr/local/lib/python3.8/dist-packages (3.4.1)
Requirement already satisfied: direct-redis in /usr/local/lib/python3.8/dist-packages (0.3.1)
Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.8/dist-packages (from pandas) (2023.3.post1)
Requirement already satisfied: numpy>=1.20.3; python_version < "3.10" in /usr/local/lib/python3.8/dist-packages (from pandas) (1.24.4)
Requirement already satisfied: tzdata>=2022.1 in /usr/local/lib/python3.8/dist-packages (from pandas) (2023.3)
Requirement already satisfied: python-dateutil>=2.8.2 in /usr/local/lib/python3.8/dist-packages (from pandas) (2.8.2)
Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.8/dist-packages (from python-dateutil>=2.8.2->pandas) (1.16.0)
All dependencies are installed
Running /platform/source/s01/v4_enggpt/m00022211/m00022211_q12_t0.sh
0% [Working]            Hit:1 http://ports.ubuntu.com/ubuntu-ports focal InRelease
0% [Working]            Get:2 http://ports.ubuntu.com/ubuntu-ports focal-updates InRelease [114 kB]
0% [2 InRelease 1239 B/114 kB 1%]0% [2 InRelease 15.7 kB/114 kB 14%]                                   0% [Working]0% [Waiting for headers]                        Hit:3 http://ports.ubuntu.com/ubuntu-ports focal-backports InRelease
                        0% [Working]0% [Waiting for headers]                        Hit:4 http://ports.ubuntu.com/ubuntu-ports focal-security InRelease
                        0% [Working]21% [Waiting for headers]                         Get:5 http://ports.ubuntu.com/ubuntu-ports focal-updates/universe arm64 Packages [1328 kB]
21% [5 Packages 5581 B/1328 kB 0%]                                  46% [Waiting for headers]                         Get:6 http://ports.ubuntu.com/ubuntu-ports focal-updates/main arm64 Packages [2804 kB]
48% [6 Packages 110 kB/2804 kB 4%]49% [5 Packages store 0 B] [6 Packages 144 kB/2804 kB 5%]                                                         63% [6 Packages 889 kB/2804 kB 32%]                                   99% [Working]99% [6 Packages store 0 B]                          100% [Working]              Fetched 4246 kB in 2s (2682 kB/s)
Reading package lists... 0%Reading package lists... 0%Reading package lists... 0%Reading package lists... 6%Reading package lists... 6%Reading package lists... 6%Reading package lists... 6%Reading package lists... 60%Reading package lists... 60%Reading package lists... 61%Reading package lists... 61%Reading package lists... 74%Reading package lists... 74%Reading package lists... 75%Reading package lists... 75%Reading package lists... 81%Reading package lists... 81%Reading package lists... 81%Reading package lists... 81%Reading package lists... 82%Reading package lists... 82%Reading package lists... 82%Reading package lists... 82%Reading package lists... 94%Reading package lists... 94%Reading package lists... 94%Reading package lists... 94%Reading package lists... 99%Reading package lists... 99%Reading package lists... 99%Reading package lists... 99%Reading package lists... Done
Reading package lists... 0%Reading package lists... 0%Reading package lists... 0%Reading package lists... 6%Reading package lists... 6%Reading package lists... 6%Reading package lists... 6%Reading package lists... 60%Reading package lists... 60%Reading package lists... 61%Reading package lists... 61%Reading package lists... 74%Reading package lists... 74%Reading package lists... 75%Reading package lists... 75%Reading package lists... 81%Reading package lists... 81%Reading package lists... 81%Reading package lists... 81%Reading package lists... 82%Reading package lists... 82%Reading package lists... 82%Reading package lists... 82%Reading package lists... 94%Reading package lists... 94%Reading package lists... 94%Reading package lists... 94%Reading package lists... 99%Reading package lists... 99%Reading package lists... 99%Reading package lists... 99%Reading package lists... Done
Building dependency tree... 0%Building dependency tree... 0%Building dependency tree... 50%Building dependency tree... 50%Building dependency tree       
Reading state information... 0%Reading state information... 0%Reading state information... Done
Calculating upgrade... 0%Calculating upgrade... 50%Calculating upgrade... Done
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Reading package lists... 0%Reading package lists... 0%Reading package lists... 0%Reading package lists... 6%Reading package lists... 6%Reading package lists... 6%Reading package lists... 6%Reading package lists... 60%Reading package lists... 60%Reading package lists... 61%Reading package lists... 61%Reading package lists... 74%Reading package lists... 74%Reading package lists... 75%Reading package lists... 75%Reading package lists... 81%Reading package lists... 81%Reading package lists... 81%Reading package lists... 81%Reading package lists... 82%Reading package lists... 82%Reading package lists... 82%Reading package lists... 82%Reading package lists... 94%Reading package lists... 94%Reading package lists... 94%Reading package lists... 94%Reading package lists... 99%Reading package lists... 99%Reading package lists... 99%Reading package lists... 99%Reading package lists... Done
Building dependency tree... 0%Building dependency tree... 0%Building dependency tree... 50%Building dependency tree... 50%Building dependency tree       
Reading state information... 0%Reading state information... 0%Reading state information... Done
python3-pip is already the newest version (20.0.2-5ubuntu1.10).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Requirement already satisfied: pymongo in /usr/local/lib/python3.8/dist-packages (4.6.0)
Requirement already satisfied: dnspython<3.0.0,>=1.16.0 in /usr/local/lib/python3.8/dist-packages (from pymongo) (2.4.2)
Running /platform/source/s01/v4_enggpt/m00022211/m00022211_q13_t0.sh
/platform/source/s01/v4_enggpt/m00022211/m00022211_q13_t0.sh: line 4: sudo: command not found
/platform/source/s01/v4_enggpt/m00022211/m00022211_q13_t0.sh: line 7: sudo: command not found
Requirement already satisfied: pymongo in /usr/local/lib/python3.8/dist-packages (4.6.0)
Requirement already satisfied: pandas in /usr/local/lib/python3.8/dist-packages (2.0.3)
Requirement already satisfied: dnspython<3.0.0,>=1.16.0 in /usr/local/lib/python3.8/dist-packages (from pymongo) (2.4.2)
Requirement already satisfied: numpy>=1.20.3; python_version < "3.10" in /usr/local/lib/python3.8/dist-packages (from pandas) (1.24.4)
Requirement already satisfied: python-dateutil>=2.8.2 in /usr/local/lib/python3.8/dist-packages (from pandas) (2.8.2)
Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.8/dist-packages (from pandas) (2023.3.post1)
Requirement already satisfied: tzdata>=2022.1 in /usr/local/lib/python3.8/dist-packages (from pandas) (2023.3)
Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.8/dist-packages (from python-dateutil>=2.8.2->pandas) (1.16.0)
Dependencies for Python script installed.
Running /platform/source/s01/v4_enggpt/m00022211/m00022211_q14_t0.sh
0% [Working]            Hit:1 http://ports.ubuntu.com/ubuntu-ports focal InRelease
0% [Working]0% [Waiting for headers]                        Hit:2 http://ports.ubuntu.com/ubuntu-ports focal-updates InRelease
                        0% [Working]0% [Waiting for headers]                        Hit:3 http://ports.ubuntu.com/ubuntu-ports focal-backports InRelease
                        0% [Working]0% [Waiting for headers]                        Hit:4 http://ports.ubuntu.com/ubuntu-ports focal-security InRelease
                        0% [Working]20% [Working]             Reading package lists... 0%Reading package lists... 0%Reading package lists... 0%Reading package lists... 6%Reading package lists... 6%Reading package lists... 6%Reading package lists... 6%Reading package lists... 60%Reading package lists... 60%Reading package lists... 61%Reading package lists... 61%Reading package lists... 74%Reading package lists... 74%Reading package lists... 75%Reading package lists... 75%Reading package lists... 81%Reading package lists... 81%Reading package lists... 81%Reading package lists... 81%Reading package lists... 82%Reading package lists... 82%Reading package lists... 82%Reading package lists... 82%Reading package lists... 94%Reading package lists... 94%Reading package lists... 94%Reading package lists... 94%Reading package lists... 99%Reading package lists... 99%Reading package lists... 99%Reading package lists... 99%Reading package lists... Done
Reading package lists... 0%Reading package lists... 0%Reading package lists... 0%Reading package lists... 6%Reading package lists... 6%Reading package lists... 6%Reading package lists... 6%Reading package lists... 60%Reading package lists... 60%Reading package lists... 61%Reading package lists... 61%Reading package lists... 74%Reading package lists... 74%Reading package lists... 75%Reading package lists... 75%Reading package lists... 81%Reading package lists... 81%Reading package lists... 81%Reading package lists... 81%Reading package lists... 82%Reading package lists... 82%Reading package lists... 82%Reading package lists... 82%Reading package lists... 94%Reading package lists... 94%Reading package lists... 94%Reading package lists... 94%Reading package lists... 99%Reading package lists... 99%Reading package lists... 99%Reading package lists... 99%Reading package lists... Done
Building dependency tree... 0%Building dependency tree... 0%Building dependency tree... 50%Building dependency tree... 50%Building dependency tree       
Reading state information... 0%Reading state information... 0%Reading state information... Done
python3 is already the newest version (3.8.2-0ubuntu2).
python3 set to manually installed.
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Reading package lists... 0%Reading package lists... 0%Reading package lists... 0%Reading package lists... 6%Reading package lists... 6%Reading package lists... 6%Reading package lists... 6%Reading package lists... 60%Reading package lists... 60%Reading package lists... 61%Reading package lists... 61%Reading package lists... 74%Reading package lists... 74%Reading package lists... 75%Reading package lists... 75%Reading package lists... 81%Reading package lists... 81%Reading package lists... 81%Reading package lists... 81%Reading package lists... 82%Reading package lists... 82%Reading package lists... 82%Reading package lists... 82%Reading package lists... 94%Reading package lists... 94%Reading package lists... 94%Reading package lists... 94%Reading package lists... 99%Reading package lists... 99%Reading package lists... 99%Reading package lists... 99%Reading package lists... Done
Building dependency tree... 0%Building dependency tree... 0%Building dependency tree... 50%Building dependency tree... 50%Building dependency tree       
Reading state information... 0%Reading state information... 0%Reading state information... Done
python3-pip is already the newest version (20.0.2-5ubuntu1.10).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Requirement already satisfied: pymysql in /usr/local/lib/python3.8/dist-packages (1.1.0)
Requirement already satisfied: pymongo in /usr/local/lib/python3.8/dist-packages (4.6.0)
Requirement already satisfied: dnspython<3.0.0,>=1.16.0 in /usr/local/lib/python3.8/dist-packages (from pymongo) (2.4.2)
Running /platform/source/s01/v4_enggpt/m00022211/m00022211_q15_t0.sh
Running /platform/source/s01/v4_enggpt/m00022211/m00022211_q16_t0.sh
/platform/source/s01/v4_enggpt/m00022211/m00022211_q16_t0.sh: line 5: sudo: command not found
/platform/source/s01/v4_enggpt/m00022211/m00022211_q16_t0.sh: line 6: sudo: command not found
Requirement already satisfied: pymysql in /usr/local/lib/python3.8/dist-packages (1.1.0)
Requirement already satisfied: pandas in /usr/local/lib/python3.8/dist-packages (2.0.3)
Requirement already satisfied: direct-redis in /usr/local/lib/python3.8/dist-packages (0.3.1)
Requirement already satisfied: tzdata>=2022.1 in /usr/local/lib/python3.8/dist-packages (from pandas) (2023.3)
Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.8/dist-packages (from pandas) (2023.3.post1)
Requirement already satisfied: python-dateutil>=2.8.2 in /usr/local/lib/python3.8/dist-packages (from pandas) (2.8.2)
Requirement already satisfied: numpy>=1.20.3; python_version < "3.10" in /usr/local/lib/python3.8/dist-packages (from pandas) (1.24.4)
Requirement already satisfied: redis==3.4.1 in /usr/local/lib/python3.8/dist-packages (from direct-redis) (3.4.1)
Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.8/dist-packages (from python-dateutil>=2.8.2->pandas) (1.16.0)
Running /platform/source/s01/v4_enggpt/m00022211/m00022211_q17_t0.sh
0% [Working]            Hit:1 http://ports.ubuntu.com/ubuntu-ports focal InRelease
0% [Working]0% [Waiting for headers]                        Hit:2 http://ports.ubuntu.com/ubuntu-ports focal-updates InRelease
                        0% [Working]0% [Waiting for headers]                        Hit:3 http://ports.ubuntu.com/ubuntu-ports focal-backports InRelease
                        0% [Working]0% [Waiting for headers]                        Hit:4 http://ports.ubuntu.com/ubuntu-ports focal-security InRelease
                        0% [Working]20% [Working]             Reading package lists... 0%Reading package lists... 0%Reading package lists... 0%Reading package lists... 6%Reading package lists... 6%Reading package lists... 6%Reading package lists... 6%Reading package lists... 60%Reading package lists... 60%Reading package lists... 61%Reading package lists... 61%Reading package lists... 74%Reading package lists... 74%Reading package lists... 75%Reading package lists... 75%Reading package lists... 81%Reading package lists... 81%Reading package lists... 81%Reading package lists... 81%Reading package lists... 82%Reading package lists... 82%Reading package lists... 82%Reading package lists... 82%Reading package lists... 94%Reading package lists... 94%Reading package lists... 94%Reading package lists... 94%Reading package lists... 99%Reading package lists... 99%Reading package lists... 99%Reading package lists... 99%Reading package lists... Done
Reading package lists... 0%Reading package lists... 0%Reading package lists... 0%Reading package lists... 6%Reading package lists... 6%Reading package lists... 6%Reading package lists... 6%Reading package lists... 60%Reading package lists... 60%Reading package lists... 61%Reading package lists... 61%Reading package lists... 74%Reading package lists... 74%Reading package lists... 75%Reading package lists... 75%Reading package lists... 81%Reading package lists... 81%Reading package lists... 81%Reading package lists... 81%Reading package lists... 82%Reading package lists... 82%Reading package lists... 82%Reading package lists... 82%Reading package lists... 94%Reading package lists... 94%Reading package lists... 94%Reading package lists... 94%Reading package lists... 99%Reading package lists... 99%Reading package lists... 99%Reading package lists... 99%Reading package lists... Done
Building dependency tree... 0%Building dependency tree... 0%Building dependency tree... 50%Building dependency tree... 50%Building dependency tree       
Reading state information... 0%Reading state information... 0%Reading state information... Done
python3 is already the newest version (3.8.2-0ubuntu2).
python3-pip is already the newest version (20.0.2-5ubuntu1.10).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Requirement already satisfied: pymysql in /usr/local/lib/python3.8/dist-packages (1.1.0)
Requirement already satisfied: pymongo in /usr/local/lib/python3.8/dist-packages (4.6.0)
Requirement already satisfied: dnspython<3.0.0,>=1.16.0 in /usr/local/lib/python3.8/dist-packages (from pymongo) (2.4.2)
Running /platform/source/s01/v4_enggpt/m00022211/m00022211_q18_t0.sh
/platform/source/s01/v4_enggpt/m00022211/m00022211_q18_t0.sh: line 5: sudo: command not found
/platform/source/s01/v4_enggpt/m00022211/m00022211_q18_t0.sh: line 8: sudo: command not found
Requirement already satisfied: pymongo in /usr/local/lib/python3.8/dist-packages (4.6.0)
Requirement already satisfied: pandas in /usr/local/lib/python3.8/dist-packages (2.0.3)
Requirement already satisfied: direct_redis in /usr/local/lib/python3.8/dist-packages (0.3.1)
Requirement already satisfied: dnspython<3.0.0,>=1.16.0 in /usr/local/lib/python3.8/dist-packages (from pymongo) (2.4.2)
Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.8/dist-packages (from pandas) (2023.3.post1)
Requirement already satisfied: numpy>=1.20.3; python_version < "3.10" in /usr/local/lib/python3.8/dist-packages (from pandas) (1.24.4)
Requirement already satisfied: python-dateutil>=2.8.2 in /usr/local/lib/python3.8/dist-packages (from pandas) (2.8.2)
Requirement already satisfied: tzdata>=2022.1 in /usr/local/lib/python3.8/dist-packages (from pandas) (2023.3)
Requirement already satisfied: redis==3.4.1 in /usr/local/lib/python3.8/dist-packages (from direct_redis) (3.4.1)
Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.8/dist-packages (from python-dateutil>=2.8.2->pandas) (1.16.0)
Running /platform/source/s01/v4_enggpt/m00022211/m00022211_q19_t0.sh
/platform/source/s01/v4_enggpt/m00022211/m00022211_q19_t0.sh: line 5: sudo: command not found
/platform/source/s01/v4_enggpt/m00022211/m00022211_q19_t0.sh: line 8: sudo: command not found
Requirement already satisfied: pymysql in /usr/local/lib/python3.8/dist-packages (1.1.0)
Requirement already satisfied: pymongo in /usr/local/lib/python3.8/dist-packages (4.6.0)
Requirement already satisfied: dnspython<3.0.0,>=1.16.0 in /usr/local/lib/python3.8/dist-packages (from pymongo) (2.4.2)
Running /platform/source/s01/v4_enggpt/m00022211/m00022211_q1_t0.sh
Running /platform/source/s01/v4_enggpt/m00022211/m00022211_q20_t0.sh
0% [Working]            Hit:1 http://ports.ubuntu.com/ubuntu-ports focal InRelease
0% [Working]0% [Waiting for headers]                        Hit:2 http://ports.ubuntu.com/ubuntu-ports focal-updates InRelease
                        0% [Working]0% [Waiting for headers]                        Hit:3 http://ports.ubuntu.com/ubuntu-ports focal-backports InRelease
0% [Waiting for headers]0% [Waiting for headers]                        Hit:4 http://ports.ubuntu.com/ubuntu-ports focal-security InRelease
                        0% [Working]20% [Working]             Reading package lists... 0%Reading package lists... 0%Reading package lists... 0%Reading package lists... 6%Reading package lists... 6%Reading package lists... 6%Reading package lists... 6%Reading package lists... 60%Reading package lists... 60%Reading package lists... 61%Reading package lists... 61%Reading package lists... 74%Reading package lists... 74%Reading package lists... 75%Reading package lists... 75%Reading package lists... 81%Reading package lists... 81%Reading package lists... 81%Reading package lists... 81%Reading package lists... 82%Reading package lists... 82%Reading package lists... 82%Reading package lists... 82%Reading package lists... 94%Reading package lists... 94%Reading package lists... 94%Reading package lists... 94%Reading package lists... 99%Reading package lists... 99%Reading package lists... 99%Reading package lists... 99%Reading package lists... Done
Reading package lists... 0%Reading package lists... 0%Reading package lists... 0%Reading package lists... 6%Reading package lists... 6%Reading package lists... 6%Reading package lists... 6%Reading package lists... 60%Reading package lists... 60%Reading package lists... 61%Reading package lists... 61%Reading package lists... 74%Reading package lists... 74%Reading package lists... 75%Reading package lists... 75%Reading package lists... 81%Reading package lists... 81%Reading package lists... 81%Reading package lists... 81%Reading package lists... 82%Reading package lists... 82%Reading package lists... 82%Reading package lists... 82%Reading package lists... 94%Reading package lists... 94%Reading package lists... 94%Reading package lists... 94%Reading package lists... 99%Reading package lists... 99%Reading package lists... 99%Reading package lists... 99%Reading package lists... Done
Building dependency tree... 0%Building dependency tree... 0%Building dependency tree... 50%Building dependency tree... 50%Building dependency tree       
Reading state information... 0%Reading state information... 0%Reading state information... Done
python3 is already the newest version (3.8.2-0ubuntu2).
python3-pip is already the newest version (20.0.2-5ubuntu1.10).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Collecting pip
  Downloading pip-23.3.1-py3-none-any.whl (2.1 MB)
[?25l[K     |▏                               | 10 kB 50.8 MB/s eta 0:00:01[K     |▎                               | 20 kB 5.8 MB/s eta 0:00:01[K     |▌                               | 30 kB 8.5 MB/s eta 0:00:01[K     |▋                               | 40 kB 3.7 MB/s eta 0:00:01[K     |▊                               | 51 kB 4.6 MB/s eta 0:00:01[K     |█                               | 61 kB 5.5 MB/s eta 0:00:01[K     |█                               | 71 kB 5.9 MB/s eta 0:00:01[K     |█▎                              | 81 kB 5.1 MB/s eta 0:00:01[K     |█▍                              | 92 kB 5.8 MB/s eta 0:00:01[K     |█▌                              | 102 kB 6.3 MB/s eta 0:00:01[K     |█▊                              | 112 kB 6.3 MB/s eta 0:00:01[K     |█▉                              | 122 kB 6.3 MB/s eta 0:00:01[K     |██                              | 133 kB 6.3 MB/s eta 0:00:01[K     |██▏                             | 143 kB 6.3 MB/s eta 0:00:01[K     |██▎                             | 153 kB 6.3 MB/s eta 0:00:01[K     |██▌                             | 163 kB 6.3 MB/s eta 0:00:01[K     |██▋                             | 174 kB 6.3 MB/s eta 0:00:01[K     |██▉                             | 184 kB 6.3 MB/s eta 0:00:01[K     |███                             | 194 kB 6.3 MB/s eta 0:00:01[K     |███                             | 204 kB 6.3 MB/s eta 0:00:01[K     |███▎                            | 215 kB 6.3 MB/s eta 0:00:01[K     |███▍                            | 225 kB 6.3 MB/s eta 0:00:01[K     |███▋                            | 235 kB 6.3 MB/s eta 0:00:01[K     |███▊                            | 245 kB 6.3 MB/s eta 0:00:01[K     |███▉                            | 256 kB 6.3 MB/s eta 0:00:01[K     |████                            | 266 kB 6.3 MB/s eta 0:00:01[K     |████▏                           | 276 kB 6.3 MB/s eta 0:00:01[K     |████▍                           | 286 kB 6.3 MB/s eta 0:00:01[K     |████▌                           | 296 kB 6.3 MB/s eta 0:00:01[K     |████▋                           | 307 kB 6.3 MB/s eta 0:00:01[K     |████▉                           | 317 kB 6.3 MB/s eta 0:00:01[K     |█████                           | 327 kB 6.3 MB/s eta 0:00:01[K     |█████▏                          | 337 kB 6.3 MB/s eta 0:00:01[K     |█████▎                          | 348 kB 6.3 MB/s eta 0:00:01[K     |█████▍                          | 358 kB 6.3 MB/s eta 0:00:01[K     |█████▋                          | 368 kB 6.3 MB/s eta 0:00:01[K     |█████▊                          | 378 kB 6.3 MB/s eta 0:00:01[K     |██████                          | 389 kB 6.3 MB/s eta 0:00:01[K     |██████                          | 399 kB 6.3 MB/s eta 0:00:01[K     |██████▏                         | 409 kB 6.3 MB/s eta 0:00:01[K     |██████▍                         | 419 kB 6.3 MB/s eta 0:00:01[K     |██████▌                         | 430 kB 6.3 MB/s eta 0:00:01[K     |██████▊                         | 440 kB 6.3 MB/s eta 0:00:01[K     |██████▉                         | 450 kB 6.3 MB/s eta 0:00:01[K     |███████                         | 460 kB 6.3 MB/s eta 0:00:01[K     |███████▏                        | 471 kB 6.3 MB/s eta 0:00:01[K     |███████▎                        | 481 kB 6.3 MB/s eta 0:00:01[K     |███████▌                        | 491 kB 6.3 MB/s eta 0:00:01[K     |███████▋                        | 501 kB 6.3 MB/s eta 0:00:01[K     |███████▊                        | 512 kB 6.3 MB/s eta 0:00:01[K     |████████                        | 522 kB 6.3 MB/s eta 0:00:01[K     |████████                        | 532 kB 6.3 MB/s eta 0:00:01[K     |████████▎                       | 542 kB 6.3 MB/s eta 0:00:01[K     |████████▍                       | 552 kB 6.3 MB/s eta 0:00:01[K     |████████▌                       | 563 kB 6.3 MB/s eta 0:00:01[K     |████████▊                       | 573 kB 6.3 MB/s eta 0:00:01[K     |████████▉                       | 583 kB 6.3 MB/s eta 0:00:01[K     |█████████                       | 593 kB 6.3 MB/s eta 0:00:01[K     |█████████▏                      | 604 kB 6.3 MB/s eta 0:00:01[K     |█████████▎                      | 614 kB 6.3 MB/s eta 0:00:01[K     |█████████▌                      | 624 kB 6.3 MB/s eta 0:00:01[K     |█████████▋                      | 634 kB 6.3 MB/s eta 0:00:01[K     |█████████▉                      | 645 kB 6.3 MB/s eta 0:00:01[K     |██████████                      | 655 kB 6.3 MB/s eta 0:00:01[K     |██████████                      | 665 kB 6.3 MB/s eta 0:00:01[K     |██████████▎                     | 675 kB 6.3 MB/s eta 0:00:01[K     |██████████▍                     | 686 kB 6.3 MB/s eta 0:00:01[K     |██████████▋                     | 696 kB 6.3 MB/s eta 0:00:01[K     |██████████▊                     | 706 kB 6.3 MB/s eta 0:00:01[K     |██████████▉                     | 716 kB 6.3 MB/s eta 0:00:01[K     |███████████                     | 727 kB 6.3 MB/s eta 0:00:01[K     |███████████▏                    | 737 kB 6.3 MB/s eta 0:00:01[K     |███████████▍                    | 747 kB 6.3 MB/s eta 0:00:01[K     |███████████▌                    | 757 kB 6.3 MB/s eta 0:00:01[K     |███████████▋                    | 768 kB 6.3 MB/s eta 0:00:01[K     |███████████▉                    | 778 kB 6.3 MB/s eta 0:00:01[K     |████████████                    | 788 kB 6.3 MB/s eta 0:00:01[K     |████████████▏                   | 798 kB 6.3 MB/s eta 0:00:01[K     |████████████▎                   | 808 kB 6.3 MB/s eta 0:00:01[K     |████████████▍                   | 819 kB 6.3 MB/s eta 0:00:01[K     |████████████▋                   | 829 kB 6.3 MB/s eta 0:00:01[K     |████████████▊                   | 839 kB 6.3 MB/s eta 0:00:01[K     |█████████████                   | 849 kB 6.3 MB/s eta 0:00:01[K     |█████████████                   | 860 kB 6.3 MB/s eta 0:00:01[K     |█████████████▏                  | 870 kB 6.3 MB/s eta 0:00:01[K     |█████████████▍                  | 880 kB 6.3 MB/s eta 0:00:01[K     |█████████████▌                  | 890 kB 6.3 MB/s eta 0:00:01[K     |█████████████▊                  | 901 kB 6.3 MB/s eta 0:00:01[K     |█████████████▉                  | 911 kB 6.3 MB/s eta 0:00:01[K     |██████████████                  | 921 kB 6.3 MB/s eta 0:00:01[K     |██████████████▏                 | 931 kB 6.3 MB/s eta 0:00:01[K     |██████████████▎                 | 942 kB 6.3 MB/s eta 0:00:01[K     |██████████████▌                 | 952 kB 6.3 MB/s eta 0:00:01[K     |██████████████▋                 | 962 kB 6.3 MB/s eta 0:00:01[K     |██████████████▊                 | 972 kB 6.3 MB/s eta 0:00:01[K     |███████████████                 | 983 kB 6.3 MB/s eta 0:00:01[K     |███████████████                 | 993 kB 6.3 MB/s eta 0:00:01[K     |███████████████▎                | 1.0 MB 6.3 MB/s eta 0:00:01[K     |███████████████▍                | 1.0 MB 6.3 MB/s eta 0:00:01[K     |███████████████▌                | 1.0 MB 6.3 MB/s eta 0:00:01[K     |███████████████▊                | 1.0 MB 6.3 MB/s eta 0:00:01[K     |███████████████▉                | 1.0 MB 6.3 MB/s eta 0:00:01[K     |████████████████                | 1.1 MB 6.3 MB/s eta 0:00:01[K     |████████████████▏               | 1.1 MB 6.3 MB/s eta 0:00:01[K     |████████████████▎               | 1.1 MB 6.3 MB/s eta 0:00:01[K     |████████████████▌               | 1.1 MB 6.3 MB/s eta 0:00:01[K     |████████████████▋               | 1.1 MB 6.3 MB/s eta 0:00:01[K     |████████████████▉               | 1.1 MB 6.3 MB/s eta 0:00:01[K     |█████████████████               | 1.1 MB 6.3 MB/s eta 0:00:01[K     |█████████████████               | 1.1 MB 6.3 MB/s eta 0:00:01[K     |█████████████████▎              | 1.1 MB 6.3 MB/s eta 0:00:01[K     |█████████████████▍              | 1.1 MB 6.3 MB/s eta 0:00:01[K     |█████████████████▋              | 1.2 MB 6.3 MB/s eta 0:00:01[K     |█████████████████▊              | 1.2 MB 6.3 MB/s eta 0:00:01[K     |█████████████████▉              | 1.2 MB 6.3 MB/s eta 0:00:01[K     |██████████████████              | 1.2 MB 6.3 MB/s eta 0:00:01[K     |██████████████████▏             | 1.2 MB 6.3 MB/s eta 0:00:01[K     |██████████████████▍             | 1.2 MB 6.3 MB/s eta 0:00:01[K     |██████████████████▌             | 1.2 MB 6.3 MB/s eta 0:00:01[K     |██████████████████▋             | 1.2 MB 6.3 MB/s eta 0:00:01[K     |██████████████████▉             | 1.2 MB 6.3 MB/s eta 0:00:01[K     |███████████████████             | 1.2 MB 6.3 MB/s eta 0:00:01[K     |███████████████████▏            | 1.3 MB 6.3 MB/s eta 0:00:01[K     |███████████████████▎            | 1.3 MB 6.3 MB/s eta 0:00:01[K     |███████████████████▍            | 1.3 MB 6.3 MB/s eta 0:00:01[K     |███████████████████▋            | 1.3 MB 6.3 MB/s eta 0:00:01[K     |███████████████████▊            | 1.3 MB 6.3 MB/s eta 0:00:01[K     |████████████████████            | 1.3 MB 6.3 MB/s eta 0:00:01[K     |████████████████████            | 1.3 MB 6.3 MB/s eta 0:00:01[K     |████████████████████▏           | 1.3 MB 6.3 MB/s eta 0:00:01[K     |████████████████████▍           | 1.3 MB 6.3 MB/s eta 0:00:01[K     |████████████████████▌           | 1.4 MB 6.3 MB/s eta 0:00:01[K     |████████████████████▊           | 1.4 MB 6.3 MB/s eta 0:00:01[K     |████████████████████▉           | 1.4 MB 6.3 MB/s eta 0:00:01[K     |█████████████████████           | 1.4 MB 6.3 MB/s eta 0:00:01[K     |█████████████████████▏          | 1.4 MB 6.3 MB/s eta 0:00:01[K     |█████████████████████▎          | 1.4 MB 6.3 MB/s eta 0:00:01[K     |█████████████████████▌          | 1.4 MB 6.3 MB/s eta 0:00:01[K     |█████████████████████▋          | 1.4 MB 6.3 MB/s eta 0:00:01[K     |█████████████████████▊          | 1.4 MB 6.3 MB/s eta 0:00:01[K     |██████████████████████          | 1.4 MB 6.3 MB/s eta 0:00:01[K     |██████████████████████          | 1.5 MB 6.3 MB/s eta 0:00:01[K     |██████████████████████▎         | 1.5 MB 6.3 MB/s eta 0:00:01[K     |██████████████████████▍         | 1.5 MB 6.3 MB/s eta 0:00:01[K     |██████████████████████▌         | 1.5 MB 6.3 MB/s eta 0:00:01[K     |██████████████████████▊         | 1.5 MB 6.3 MB/s eta 0:00:01[K     |██████████████████████▉         | 1.5 MB 6.3 MB/s eta 0:00:01[K     |███████████████████████         | 1.5 MB 6.3 MB/s eta 0:00:01[K     |███████████████████████▏        | 1.5 MB 6.3 MB/s eta 0:00:01[K     |███████████████████████▎        | 1.5 MB 6.3 MB/s eta 0:00:01[K     |███████████████████████▌        | 1.5 MB 6.3 MB/s eta 0:00:01[K     |███████████████████████▋        | 1.6 MB 6.3 MB/s eta 0:00:01[K     |███████████████████████▉        | 1.6 MB 6.3 MB/s eta 0:00:01[K     |████████████████████████        | 1.6 MB 6.3 MB/s eta 0:00:01[K     |████████████████████████        | 1.6 MB 6.3 MB/s eta 0:00:01[K     |████████████████████████▎       | 1.6 MB 6.3 MB/s eta 0:00:01[K     |████████████████████████▍       | 1.6 MB 6.3 MB/s eta 0:00:01[K     |████████████████████████▋       | 1.6 MB 6.3 MB/s eta 0:00:01[K     |████████████████████████▊       | 1.6 MB 6.3 MB/s eta 0:00:01[K     |████████████████████████▉       | 1.6 MB 6.3 MB/s eta 0:00:01[K     |█████████████████████████       | 1.6 MB 6.3 MB/s eta 0:00:01[K     |█████████████████████████▏      | 1.7 MB 6.3 MB/s eta 0:00:01[K     |█████████████████████████▍      | 1.7 MB 6.3 MB/s eta 0:00:01[K     |█████████████████████████▌      | 1.7 MB 6.3 MB/s eta 0:00:01[K     |█████████████████████████▋      | 1.7 MB 6.3 MB/s eta 0:00:01[K     |█████████████████████████▉      | 1.7 MB 6.3 MB/s eta 0:00:01[K     |██████████████████████████      | 1.7 MB 6.3 MB/s eta 0:00:01[K     |██████████████████████████▏     | 1.7 MB 6.3 MB/s eta 0:00:01[K     |██████████████████████████▎     | 1.7 MB 6.3 MB/s eta 0:00:01[K     |██████████████████████████▍     | 1.7 MB 6.3 MB/s eta 0:00:01[K     |██████████████████████████▋     | 1.8 MB 6.3 MB/s eta 0:00:01[K     |██████████████████████████▊     | 1.8 MB 6.3 MB/s eta 0:00:01[K     |███████████████████████████     | 1.8 MB 6.3 MB/s eta 0:00:01[K     |███████████████████████████     | 1.8 MB 6.3 MB/s eta 0:00:01[K     |███████████████████████████▏    | 1.8 MB 6.3 MB/s eta 0:00:01[K     |███████████████████████████▍    | 1.8 MB 6.3 MB/s eta 0:00:01[K     |███████████████████████████▌    | 1.8 MB 6.3 MB/s eta 0:00:01[K     |███████████████████████████▊    | 1.8 MB 6.3 MB/s eta 0:00:01[K     |███████████████████████████▉    | 1.8 MB 6.3 MB/s eta 0:00:01[K     |████████████████████████████    | 1.8 MB 6.3 MB/s eta 0:00:01[K     |████████████████████████████▏   | 1.9 MB 6.3 MB/s eta 0:00:01[K     |████████████████████████████▎   | 1.9 MB 6.3 MB/s eta 0:00:01[K     |████████████████████████████▌   | 1.9 MB 6.3 MB/s eta 0:00:01[K     |████████████████████████████▋   | 1.9 MB 6.3 MB/s eta 0:00:01[K     |████████████████████████████▊   | 1.9 MB 6.3 MB/s eta 0:00:01[K     |█████████████████████████████   | 1.9 MB 6.3 MB/s eta 0:00:01[K     |█████████████████████████████   | 1.9 MB 6.3 MB/s eta 0:00:01[K     |█████████████████████████████▎  | 1.9 MB 6.3 MB/s eta 0:00:01[K     |█████████████████████████████▍  | 1.9 MB 6.3 MB/s eta 0:00:01[K     |█████████████████████████████▌  | 1.9 MB 6.3 MB/s eta 0:00:01[K     |█████████████████████████████▊  | 2.0 MB 6.3 MB/s eta 0:00:01[K     |█████████████████████████████▉  | 2.0 MB 6.3 MB/s eta 0:00:01[K     |██████████████████████████████  | 2.0 MB 6.3 MB/s eta 0:00:01[K     |██████████████████████████████▏ | 2.0 MB 6.3 MB/s eta 0:00:01[K     |██████████████████████████████▎ | 2.0 MB 6.3 MB/s eta 0:00:01[K     |██████████████████████████████▌ | 2.0 MB 6.3 MB/s eta 0:00:01[K     |██████████████████████████████▋ | 2.0 MB 6.3 MB/s eta 0:00:01[K     |██████████████████████████████▉ | 2.0 MB 6.3 MB/s eta 0:00:01[K     |███████████████████████████████ | 2.0 MB 6.3 MB/s eta 0:00:01[K     |███████████████████████████████ | 2.0 MB 6.3 MB/s eta 0:00:01[K     |███████████████████████████████▎| 2.1 MB 6.3 MB/s eta 0:00:01[K     |███████████████████████████████▍| 2.1 MB 6.3 MB/s eta 0:00:01[K     |███████████████████████████████▋| 2.1 MB 6.3 MB/s eta 0:00:01[K     |███████████████████████████████▊| 2.1 MB 6.3 MB/s eta 0:00:01[K     |███████████████████████████████▉| 2.1 MB 6.3 MB/s eta 0:00:01[K     |████████████████████████████████| 2.1 MB 6.3 MB/s 
[?25hInstalling collected packages: pip
  Attempting uninstall: pip
    Found existing installation: pip 20.0.2
    Not uninstalling pip at /usr/lib/python3/dist-packages, outside environment /usr
    Can't uninstall 'pip'. No files were found to uninstall.
Successfully installed pip-23.3.1
Collecting git+https://github.com/javatechy/direct_redis.git
  Cloning https://github.com/javatechy/direct_redis.git to /tmp/pip-req-build-1225rtu_
  Running command git clone -q https://github.com/javatechy/direct_redis.git /tmp/pip-req-build-1225rtu_
[31m  ERROR: Error [Errno 2] No such file or directory: 'git' while executing command git clone -q https://github.com/javatechy/direct_redis.git /tmp/pip-req-build-1225rtu_[0m
[31mERROR: Cannot find command 'git' - do you have 'git' installed and in your PATH?[0m
Running /platform/source/s01/v4_enggpt/m00022211/m00022211_q21_t0.sh
Running /platform/source/s01/v4_enggpt/m00022211/m00022211_q22_t0.sh
0% [Working]            Hit:1 http://ports.ubuntu.com/ubuntu-ports focal InRelease
0% [Working]0% [Waiting for headers]                        Hit:2 http://ports.ubuntu.com/ubuntu-ports focal-updates InRelease
                        0% [Working]0% [Waiting for headers]                        Hit:3 http://ports.ubuntu.com/ubuntu-ports focal-backports InRelease
                        0% [Working]0% [Waiting for headers]                        Hit:4 http://ports.ubuntu.com/ubuntu-ports focal-security InRelease
                        0% [Working]20% [Working]             Reading package lists... 0%Reading package lists... 0%Reading package lists... 0%Reading package lists... 6%Reading package lists... 6%Reading package lists... 6%Reading package lists... 6%Reading package lists... 60%Reading package lists... 60%Reading package lists... 61%Reading package lists... 61%Reading package lists... 74%Reading package lists... 74%Reading package lists... 75%Reading package lists... 75%Reading package lists... 81%Reading package lists... 81%Reading package lists... 81%Reading package lists... 81%Reading package lists... 82%Reading package lists... 82%Reading package lists... 82%Reading package lists... 82%Reading package lists... 94%Reading package lists... 94%Reading package lists... 94%Reading package lists... 94%Reading package lists... 99%Reading package lists... 99%Reading package lists... 99%Reading package lists... 99%Reading package lists... Done
Reading package lists... 0%Reading package lists... 0%Reading package lists... 0%Reading package lists... 6%Reading package lists... 6%Reading package lists... 6%Reading package lists... 6%Reading package lists... 60%Reading package lists... 60%Reading package lists... 61%Reading package lists... 61%Reading package lists... 74%Reading package lists... 74%Reading package lists... 75%Reading package lists... 75%Reading package lists... 81%Reading package lists... 81%Reading package lists... 81%Reading package lists... 81%Reading package lists... 82%Reading package lists... 82%Reading package lists... 82%Reading package lists... 82%Reading package lists... 94%Reading package lists... 94%Reading package lists... 94%Reading package lists... 94%Reading package lists... 99%Reading package lists... 99%Reading package lists... 99%Reading package lists... 99%Reading package lists... Done
Building dependency tree... 0%Building dependency tree... 0%Building dependency tree... 50%Building dependency tree... 50%Building dependency tree       
Reading state information... 0%Reading state information... 0%Reading state information... Done
Calculating upgrade... 0%Calculating upgrade... 50%Calculating upgrade... Done
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Reading package lists... 0%Reading package lists... 0%Reading package lists... 0%Reading package lists... 6%Reading package lists... 6%Reading package lists... 6%Reading package lists... 6%Reading package lists... 60%Reading package lists... 60%Reading package lists... 61%Reading package lists... 61%Reading package lists... 74%Reading package lists... 74%Reading package lists... 75%Reading package lists... 75%Reading package lists... 81%Reading package lists... 81%Reading package lists... 81%Reading package lists... 81%Reading package lists... 82%Reading package lists... 82%Reading package lists... 82%Reading package lists... 82%Reading package lists... 94%Reading package lists... 94%Reading package lists... 94%Reading package lists... 94%Reading package lists... 99%Reading package lists... 99%Reading package lists... 99%Reading package lists... 99%Reading package lists... Done
Building dependency tree... 0%Building dependency tree... 0%Building dependency tree... 50%Building dependency tree... 50%Building dependency tree       
Reading state information... 0%Reading state information... 0%Reading state information... Done
python3 is already the newest version (3.8.2-0ubuntu2).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Reading package lists... 0%Reading package lists... 0%Reading package lists... 0%Reading package lists... 6%Reading package lists... 6%Reading package lists... 6%Reading package lists... 6%Reading package lists... 60%Reading package lists... 60%Reading package lists... 61%Reading package lists... 61%Reading package lists... 74%Reading package lists... 74%Reading package lists... 75%Reading package lists... 75%Reading package lists... 81%Reading package lists... 81%Reading package lists... 81%Reading package lists... 81%Reading package lists... 82%Reading package lists... 82%Reading package lists... 82%Reading package lists... 82%Reading package lists... 94%Reading package lists... 94%Reading package lists... 94%Reading package lists... 94%Reading package lists... 99%Reading package lists... 99%Reading package lists... 99%Reading package lists... 99%Reading package lists... Done
Building dependency tree... 0%Building dependency tree... 0%Building dependency tree... 50%Building dependency tree... 50%Building dependency tree       
Reading state information... 0%Reading state information... 0%Reading state information... Done
python3-pip is already the newest version (20.0.2-5ubuntu1.10).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Requirement already satisfied: pymongo in /usr/local/lib/python3.8/dist-packages (4.6.0)
Requirement already satisfied: dnspython<3.0.0,>=1.16.0 in /usr/local/lib/python3.8/dist-packages (from pymongo) (2.4.2)
[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv[0m[33m
[0mRequirement already satisfied: pandas in /usr/local/lib/python3.8/dist-packages (2.0.3)
Requirement already satisfied: python-dateutil>=2.8.2 in /usr/local/lib/python3.8/dist-packages (from pandas) (2.8.2)
Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.8/dist-packages (from pandas) (2023.3.post1)
Requirement already satisfied: tzdata>=2022.1 in /usr/local/lib/python3.8/dist-packages (from pandas) (2023.3)
Requirement already satisfied: numpy>=1.20.3 in /usr/local/lib/python3.8/dist-packages (from pandas) (1.24.4)
Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.8/dist-packages (from python-dateutil>=2.8.2->pandas) (1.16.0)
[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv[0m[33m
[0mCollecting git+https://github.com/bluele/direct_redis.git
  Cloning https://github.com/bluele/direct_redis.git to /tmp/pip-req-build-v9l2jps5
[31m  ERROR: Error [Errno 2] No such file or directory: 'git' while executing command git version[0m[31m
[0m[31mERROR: Cannot find command 'git' - do you have 'git' installed and in your PATH?[0m[31m
[0mRunning /platform/source/s01/v4_enggpt/m00022211/m00022211_q2_t0.sh
/platform/source/s01/v4_enggpt/m00022211/m00022211_q2_t0.sh: line 4: sudo: command not found
/platform/source/s01/v4_enggpt/m00022211/m00022211_q2_t0.sh: line 7: sudo: command not found
Requirement already satisfied: pymysql in /usr/local/lib/python3.8/dist-packages (1.1.0)
Requirement already satisfied: pandas in /usr/local/lib/python3.8/dist-packages (2.0.3)
Requirement already satisfied: direct_redis in /usr/local/lib/python3.8/dist-packages (0.3.1)
Requirement already satisfied: python-dateutil>=2.8.2 in /usr/local/lib/python3.8/dist-packages (from pandas) (2.8.2)
Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.8/dist-packages (from pandas) (2023.3.post1)
Requirement already satisfied: tzdata>=2022.1 in /usr/local/lib/python3.8/dist-packages (from pandas) (2023.3)
Requirement already satisfied: numpy>=1.20.3 in /usr/local/lib/python3.8/dist-packages (from pandas) (1.24.4)
Requirement already satisfied: redis==3.4.1 in /usr/local/lib/python3.8/dist-packages (from direct_redis) (3.4.1)
Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.8/dist-packages (from python-dateutil>=2.8.2->pandas) (1.16.0)
[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv[0m[33m
[0mRunning /platform/source/s01/v4_enggpt/m00022211/m00022211_q3_t0.sh
/platform/source/s01/v4_enggpt/m00022211/m00022211_q3_t0.sh: line 5: sudo: command not found
/platform/source/s01/v4_enggpt/m00022211/m00022211_q3_t0.sh: line 6: sudo: command not found
Requirement already satisfied: pymongo in /usr/local/lib/python3.8/dist-packages (4.6.0)
Requirement already satisfied: pandas in /usr/local/lib/python3.8/dist-packages (2.0.3)
Requirement already satisfied: redis in /usr/local/lib/python3.8/dist-packages (3.4.1)
Requirement already satisfied: direct_redis in /usr/local/lib/python3.8/dist-packages (0.3.1)
Requirement already satisfied: dnspython<3.0.0,>=1.16.0 in /usr/local/lib/python3.8/dist-packages (from pymongo) (2.4.2)
Requirement already satisfied: python-dateutil>=2.8.2 in /usr/local/lib/python3.8/dist-packages (from pandas) (2.8.2)
Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.8/dist-packages (from pandas) (2023.3.post1)
Requirement already satisfied: tzdata>=2022.1 in /usr/local/lib/python3.8/dist-packages (from pandas) (2023.3)
Requirement already satisfied: numpy>=1.20.3 in /usr/local/lib/python3.8/dist-packages (from pandas) (1.24.4)
Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.8/dist-packages (from python-dateutil>=2.8.2->pandas) (1.16.0)
[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv[0m[33m
[0mRunning /platform/source/s01/v4_enggpt/m00022211/m00022211_q4_t0.sh
/platform/source/s01/v4_enggpt/m00022211/m00022211_q4_t0.sh: line 4: sudo: command not found
/platform/source/s01/v4_enggpt/m00022211/m00022211_q4_t0.sh: line 7: sudo: command not found
Requirement already satisfied: pymongo in /usr/local/lib/python3.8/dist-packages (4.6.0)
Requirement already satisfied: dnspython<3.0.0,>=1.16.0 in /usr/local/lib/python3.8/dist-packages (from pymongo) (2.4.2)
[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv[0m[33m
[0mRunning /platform/source/s01/v4_enggpt/m00022211/m00022211_q5_t0.sh
/platform/source/s01/v4_enggpt/m00022211/m00022211_q5_t0.sh: line 4: sudo: command not found
/platform/source/s01/v4_enggpt/m00022211/m00022211_q5_t0.sh: line 5: sudo: command not found
Requirement already satisfied: pymysql in /usr/local/lib/python3.8/dist-packages (1.1.0)
Requirement already satisfied: pymongo in /usr/local/lib/python3.8/dist-packages (4.6.0)
Requirement already satisfied: pandas in /usr/local/lib/python3.8/dist-packages (2.0.3)
Requirement already satisfied: direct_redis in /usr/local/lib/python3.8/dist-packages (0.3.1)
Requirement already satisfied: dnspython<3.0.0,>=1.16.0 in /usr/local/lib/python3.8/dist-packages (from pymongo) (2.4.2)
Requirement already satisfied: python-dateutil>=2.8.2 in /usr/local/lib/python3.8/dist-packages (from pandas) (2.8.2)
Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.8/dist-packages (from pandas) (2023.3.post1)
Requirement already satisfied: tzdata>=2022.1 in /usr/local/lib/python3.8/dist-packages (from pandas) (2023.3)
Requirement already satisfied: numpy>=1.20.3 in /usr/local/lib/python3.8/dist-packages (from pandas) (1.24.4)
Requirement already satisfied: redis==3.4.1 in /usr/local/lib/python3.8/dist-packages (from direct_redis) (3.4.1)
Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.8/dist-packages (from python-dateutil>=2.8.2->pandas) (1.16.0)
[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv[0m[33m
[0mRunning /platform/source/s01/v4_enggpt/m00022211/m00022211_q6_t0.sh
0% [Working]            Hit:1 http://ports.ubuntu.com/ubuntu-ports focal InRelease
0% [Working]0% [Waiting for headers]                        Hit:2 http://ports.ubuntu.com/ubuntu-ports focal-updates InRelease
0% [Waiting for headers]0% [Waiting for headers]                        Hit:3 http://ports.ubuntu.com/ubuntu-ports focal-backports InRelease
                        0% [Working]0% [Waiting for headers]                        Hit:4 http://ports.ubuntu.com/ubuntu-ports focal-security InRelease
                        0% [Working]20% [Working]             Reading package lists... 0%Reading package lists... 0%Reading package lists... 0%Reading package lists... 6%Reading package lists... 6%Reading package lists... 6%Reading package lists... 6%Reading package lists... 60%Reading package lists... 60%Reading package lists... 61%Reading package lists... 61%Reading package lists... 74%Reading package lists... 74%Reading package lists... 75%Reading package lists... 75%Reading package lists... 81%Reading package lists... 81%Reading package lists... 81%Reading package lists... 81%Reading package lists... 82%Reading package lists... 82%Reading package lists... 82%Reading package lists... 82%Reading package lists... 94%Reading package lists... 94%Reading package lists... 94%Reading package lists... 94%Reading package lists... 99%Reading package lists... 99%Reading package lists... 99%Reading package lists... 99%Reading package lists... Done
Reading package lists... 0%Reading package lists... 0%Reading package lists... 0%Reading package lists... 6%Reading package lists... 6%Reading package lists... 6%Reading package lists... 6%Reading package lists... 60%Reading package lists... 60%Reading package lists... 61%Reading package lists... 61%Reading package lists... 74%Reading package lists... 74%Reading package lists... 75%Reading package lists... 75%Reading package lists... 81%Reading package lists... 81%Reading package lists... 81%Reading package lists... 81%Reading package lists... 82%Reading package lists... 82%Reading package lists... 82%Reading package lists... 82%Reading package lists... 94%Reading package lists... 94%Reading package lists... 94%Reading package lists... 94%Reading package lists... 99%Reading package lists... 99%Reading package lists... 99%Reading package lists... 99%Reading package lists... Done
Building dependency tree... 0%Building dependency tree... 0%Building dependency tree... 50%Building dependency tree... 50%Building dependency tree       
Reading state information... 0%Reading state information... 0%Reading state information... Done
python3 is already the newest version (3.8.2-0ubuntu2).
python3-pip is already the newest version (20.0.2-5ubuntu1.10).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Requirement already satisfied: pymongo in /usr/local/lib/python3.8/dist-packages (4.6.0)
Requirement already satisfied: dnspython<3.0.0,>=1.16.0 in /usr/local/lib/python3.8/dist-packages (from pymongo) (2.4.2)
[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv[0m[33m
[0mRunning /platform/source/s01/v4_enggpt/m00022211/m00022211_q7_t0.sh
/platform/source/s01/v4_enggpt/m00022211/m00022211_q7_t0.sh: line 6: sudo: command not found
/platform/source/s01/v4_enggpt/m00022211/m00022211_q7_t0.sh: line 7: sudo: command not found
Requirement already satisfied: pymysql in /usr/local/lib/python3.8/dist-packages (1.1.0)
Requirement already satisfied: pymongo in /usr/local/lib/python3.8/dist-packages (4.6.0)
Requirement already satisfied: pandas in /usr/local/lib/python3.8/dist-packages (2.0.3)
Requirement already satisfied: direct-redis in /usr/local/lib/python3.8/dist-packages (0.3.1)
Requirement already satisfied: dnspython<3.0.0,>=1.16.0 in /usr/local/lib/python3.8/dist-packages (from pymongo) (2.4.2)
Requirement already satisfied: python-dateutil>=2.8.2 in /usr/local/lib/python3.8/dist-packages (from pandas) (2.8.2)
Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.8/dist-packages (from pandas) (2023.3.post1)
Requirement already satisfied: tzdata>=2022.1 in /usr/local/lib/python3.8/dist-packages (from pandas) (2023.3)
Requirement already satisfied: numpy>=1.20.3 in /usr/local/lib/python3.8/dist-packages (from pandas) (1.24.4)
Requirement already satisfied: redis==3.4.1 in /usr/local/lib/python3.8/dist-packages (from direct-redis) (3.4.1)
Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.8/dist-packages (from python-dateutil>=2.8.2->pandas) (1.16.0)
[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv[0m[33m
[0mRunning /platform/source/s01/v4_enggpt/m00022211/m00022211_q8_t0.sh
Requirement already satisfied: pymysql in /usr/local/lib/python3.8/dist-packages (1.1.0)
[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv[0m[33m
[0mRequirement already satisfied: pymongo in /usr/local/lib/python3.8/dist-packages (4.6.0)
Requirement already satisfied: dnspython<3.0.0,>=1.16.0 in /usr/local/lib/python3.8/dist-packages (from pymongo) (2.4.2)
[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv[0m[33m
[0mRequirement already satisfied: pandas in /usr/local/lib/python3.8/dist-packages (2.0.3)
Requirement already satisfied: python-dateutil>=2.8.2 in /usr/local/lib/python3.8/dist-packages (from pandas) (2.8.2)
Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.8/dist-packages (from pandas) (2023.3.post1)
Requirement already satisfied: tzdata>=2022.1 in /usr/local/lib/python3.8/dist-packages (from pandas) (2023.3)
Requirement already satisfied: numpy>=1.20.3 in /usr/local/lib/python3.8/dist-packages (from pandas) (1.24.4)
Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.8/dist-packages (from python-dateutil>=2.8.2->pandas) (1.16.0)
[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv[0m[33m
[0mRequirement already satisfied: direct-redis in /usr/local/lib/python3.8/dist-packages (0.3.1)
Requirement already satisfied: redis==3.4.1 in /usr/local/lib/python3.8/dist-packages (from direct-redis) (3.4.1)
[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv[0m[33m
[0mRunning /platform/source/s01/v4_enggpt/m00022211/m00022211_q9_t0.sh
0% [Working]            Hit:1 http://ports.ubuntu.com/ubuntu-ports focal InRelease
0% [Working]0% [Waiting for headers]                        Hit:2 http://ports.ubuntu.com/ubuntu-ports focal-updates InRelease
                        0% [Working]0% [Waiting for headers]                        Hit:3 http://ports.ubuntu.com/ubuntu-ports focal-backports InRelease
                        0% [Working]0% [Waiting for headers]                        Hit:4 http://ports.ubuntu.com/ubuntu-ports focal-security InRelease
                        0% [Working]20% [Working]             Reading package lists... 0%Reading package lists... 0%Reading package lists... 0%Reading package lists... 6%Reading package lists... 6%Reading package lists... 6%Reading package lists... 6%Reading package lists... 60%Reading package lists... 60%Reading package lists... 61%Reading package lists... 61%Reading package lists... 74%Reading package lists... 74%Reading package lists... 75%Reading package lists... 75%Reading package lists... 81%Reading package lists... 81%Reading package lists... 81%Reading package lists... 81%Reading package lists... 82%Reading package lists... 82%Reading package lists... 82%Reading package lists... 82%Reading package lists... 94%Reading package lists... 94%Reading package lists... 94%Reading package lists... 94%Reading package lists... 99%Reading package lists... 99%Reading package lists... 99%Reading package lists... 99%Reading package lists... Done
Reading package lists... 0%Reading package lists... 0%Reading package lists... 0%Reading package lists... 6%Reading package lists... 6%Reading package lists... 6%Reading package lists... 6%Reading package lists... 60%Reading package lists... 60%Reading package lists... 61%Reading package lists... 61%Reading package lists... 74%Reading package lists... 74%Reading package lists... 75%Reading package lists... 75%Reading package lists... 81%Reading package lists... 81%Reading package lists... 81%Reading package lists... 81%Reading package lists... 82%Reading package lists... 82%Reading package lists... 82%Reading package lists... 82%Reading package lists... 94%Reading package lists... 94%Reading package lists... 94%Reading package lists... 94%Reading package lists... 99%Reading package lists... 99%Reading package lists... 99%Reading package lists... 99%Reading package lists... Done
Building dependency tree... 0%Building dependency tree... 0%Building dependency tree... 50%Building dependency tree... 50%Building dependency tree       
Reading state information... 0%Reading state information... 0%Reading state information... Done
python3 is already the newest version (3.8.2-0ubuntu2).
python3-pip is already the newest version (20.0.2-5ubuntu1.10).
0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.
Requirement already satisfied: pymysql in /usr/local/lib/python3.8/dist-packages (1.1.0)
Requirement already satisfied: pymongo in /usr/local/lib/python3.8/dist-packages (4.6.0)
Requirement already satisfied: pandas in /usr/local/lib/python3.8/dist-packages (2.0.3)
Requirement already satisfied: direct_redis in /usr/local/lib/python3.8/dist-packages (0.3.1)
Requirement already satisfied: dnspython<3.0.0,>=1.16.0 in /usr/local/lib/python3.8/dist-packages (from pymongo) (2.4.2)
Requirement already satisfied: python-dateutil>=2.8.2 in /usr/local/lib/python3.8/dist-packages (from pandas) (2.8.2)
Requirement already satisfied: pytz>=2020.1 in /usr/local/lib/python3.8/dist-packages (from pandas) (2023.3.post1)
Requirement already satisfied: tzdata>=2022.1 in /usr/local/lib/python3.8/dist-packages (from pandas) (2023.3)
Requirement already satisfied: numpy>=1.20.3 in /usr/local/lib/python3.8/dist-packages (from pandas) (1.24.4)
Requirement already satisfied: redis==3.4.1 in /usr/local/lib/python3.8/dist-packages (from direct_redis) (3.4.1)
Requirement already satisfied: six>=1.5 in /usr/local/lib/python3.8/dist-packages (from python-dateutil>=2.8.2->pandas) (1.16.0)
[33mWARNING: Running pip as the 'root' user can result in broken permissions and conflicting behaviour with the system package manager. It is recommended to use a virtual environment instead: https://pip.pypa.io/warnings/venv[0m[33m
[0mpython /dbknitter/dbknitter/grader.py /platform/source/s01/v4_enggpt/m00022211/m00022211_q10_t0.py /platform/output/s01/v4_enggpt/m00022211/m00022211_q10_t0.csv /platform/expected/s01/q10.csv
Traceback (most recent call last):
  File "/platform/source/s01/v4_enggpt/m00022211/m00022211_q10_t0.py", line 41, in <module>
    customer_df = pd.read_json(r.get('customer'))
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/json/_json.py", line 760, in read_json
    json_reader = JsonReader(
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/json/_json.py", line 861, in __init__
    data = self._get_data_from_filepath(filepath_or_buffer)
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/json/_json.py", line 901, in _get_data_from_filepath
    self.handles = get_handle(
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/common.py", line 707, in get_handle
    if _is_binary_mode(path_or_buf, mode) and "b" not in mode:
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/common.py", line 1167, in _is_binary_mode
    return isinstance(handle, _get_binary_io_classes()) or "b" in getattr(
TypeError: argument of type 'method' is not iterable
Score.fail_execute
python /dbknitter/dbknitter/grader.py /platform/source/s01/v4_enggpt/m00022211/m00022211_q11_t0.py /platform/output/s01/v4_enggpt/m00022211/m00022211_q11_t0.csv /platform/expected/s01/q11.csv
Traceback (most recent call last):
  File "/platform/source/s01/v4_enggpt/m00022211/m00022211_q11_t0.py", line 27, in <module>
    supplier_df = pd.read_json(redis_client.get('supplier'))
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/json/_json.py", line 760, in read_json
    json_reader = JsonReader(
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/json/_json.py", line 861, in __init__
    data = self._get_data_from_filepath(filepath_or_buffer)
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/json/_json.py", line 901, in _get_data_from_filepath
    self.handles = get_handle(
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/common.py", line 707, in get_handle
    if _is_binary_mode(path_or_buf, mode) and "b" not in mode:
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/common.py", line 1167, in _is_binary_mode
    return isinstance(handle, _get_binary_io_classes()) or "b" in getattr(
TypeError: argument of type 'method' is not iterable
Score.fail_execute
python /dbknitter/dbknitter/grader.py /platform/source/s01/v4_enggpt/m00022211/m00022211_q12_t0.py /platform/output/s01/v4_enggpt/m00022211/m00022211_q12_t0.csv /platform/expected/s01/q12.csv
>>> Expected ((2, 3))
   HIGH_LINE_COUNT  LOW_LINE_COUNT L_SHIPMODE
0              647             945       MAIL
1              620             943       SHIP

>>> Actual ((2, 3))
   HIGH_LINE_COUNT  LOW_LINE_COUNT shipping_mode
0                0               0          MAIL
1                0               0          SHIP

Head row similarity: 0.8268551236749117
Score.success_approx
python /dbknitter/dbknitter/grader.py /platform/source/s01/v4_enggpt/m00022211/m00022211_q13_t0.py /platform/output/s01/v4_enggpt/m00022211/m00022211_q13_t0.csv /platform/expected/s01/q13.csv
Traceback (most recent call last):
  File "/platform/source/s01/v4_enggpt/m00022211/m00022211_q13_t0.py", line 39, in <module>
    customers_df = get_redis_data()
  File "/platform/source/s01/v4_enggpt/m00022211/m00022211_q13_t0.py", line 34, in get_redis_data
    customer_data = pd.read_json(customer_data_str)
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/json/_json.py", line 760, in read_json
    json_reader = JsonReader(
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/json/_json.py", line 861, in __init__
    data = self._get_data_from_filepath(filepath_or_buffer)
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/json/_json.py", line 901, in _get_data_from_filepath
    self.handles = get_handle(
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/common.py", line 707, in get_handle
    if _is_binary_mode(path_or_buf, mode) and "b" not in mode:
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/common.py", line 1167, in _is_binary_mode
    return isinstance(handle, _get_binary_io_classes()) or "b" in getattr(
TypeError: argument of type 'method' is not iterable
Score.fail_execute
python /dbknitter/dbknitter/grader.py /platform/source/s01/v4_enggpt/m00022211/m00022211_q14_t0.py /platform/output/s01/v4_enggpt/m00022211/m00022211_q14_t0.csv /platform/expected/s01/q14.csv
>>> Expected ((1, 1))
   PROMO_REVENUE
0      16.283856

>>> Actual ((1, 1))
   Promotional Revenue Percentage
0                       16.283856

Head row similarity: 0.44
Score.fail_wrong_output
python /dbknitter/dbknitter/grader.py /platform/source/s01/v4_enggpt/m00022211/m00022211_q15_t0.py /platform/output/s01/v4_enggpt/m00022211/m00022211_q15_t0.csv /platform/expected/s01/q15.csv
Traceback (most recent call last):
  File "/platform/source/s01/v4_enggpt/m00022211/m00022211_q15_t0.py", line 30, in <module>
    lineitem_df['REVENUE'] = lineitem_df['L_EXTENDEDPRICE'] * (1 - lineitem_df['L_DISCOUNT'])
  File "/usr/local/lib/python3.8/dist-packages/pandas/core/frame.py", line 3761, in __getitem__
    indexer = self.columns.get_loc(key)
  File "/usr/local/lib/python3.8/dist-packages/pandas/core/indexes/range.py", line 349, in get_loc
    raise KeyError(key)
KeyError: 'L_EXTENDEDPRICE'
Score.fail_execute
python /dbknitter/dbknitter/grader.py /platform/source/s01/v4_enggpt/m00022211/m00022211_q16_t0.py /platform/output/s01/v4_enggpt/m00022211/m00022211_q16_t0.csv /platform/expected/s01/q16.csv
/platform/source/s01/v4_enggpt/m00022211/m00022211_q16_t0.py:15: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  part_df = pd.read_sql(part_query, mysql_conn)
Traceback (most recent call last):
  File "/platform/source/s01/v4_enggpt/m00022211/m00022211_q16_t0.py", line 20, in <module>
    partsupp_df = pd.read_json(redis_conn.get('partsupp'))
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/json/_json.py", line 760, in read_json
    json_reader = JsonReader(
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/json/_json.py", line 861, in __init__
    data = self._get_data_from_filepath(filepath_or_buffer)
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/json/_json.py", line 901, in _get_data_from_filepath
    self.handles = get_handle(
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/common.py", line 707, in get_handle
    if _is_binary_mode(path_or_buf, mode) and "b" not in mode:
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/common.py", line 1167, in _is_binary_mode
    return isinstance(handle, _get_binary_io_classes()) or "b" in getattr(
TypeError: argument of type 'method' is not iterable
Score.fail_execute
python /dbknitter/dbknitter/grader.py /platform/source/s01/v4_enggpt/m00022211/m00022211_q17_t0.py /platform/output/s01/v4_enggpt/m00022211/m00022211_q17_t0.csv /platform/expected/s01/q17.csv
>>> Expected ((1, 1))
     AVG_YEARLY
0  31865.742857

>>> Actual ((21, 1))
    average_yearly_extended_price
0                     1224.142857
1                     1059.440000
2                     2220.171429
3                     1404.617143
4                     2124.548571
5                     1276.811429
6                      544.600000
7                     1326.392857
8                     1678.405714
9                     2853.840000
10                    1250.592857
11                    1010.357143
12                    1759.062857
13                     907.842857
14                    1263.051429
15                    1515.308571
16                     955.050000
17                     774.537143
18                     964.440000
19                    4891.015714
20                     861.514286

Score.fail_wrong_output
python /dbknitter/dbknitter/grader.py /platform/source/s01/v4_enggpt/m00022211/m00022211_q18_t0.py /platform/output/s01/v4_enggpt/m00022211/m00022211_q18_t0.csv /platform/expected/s01/q18.csv
Traceback (most recent call last):
  File "/platform/source/s01/v4_enggpt/m00022211/m00022211_q18_t0.py", line 47, in <module>
    final_result = final_result[['C_NAME', 'O_CUSTKEY', 'O_ORDERKEY', 'O_ORDERDATE', 'O_TOTALPRICE', 'L_QUANTITY']]
  File "/usr/local/lib/python3.8/dist-packages/pandas/core/frame.py", line 3767, in __getitem__
    indexer = self.columns._get_indexer_strict(key, "columns")[1]
  File "/usr/local/lib/python3.8/dist-packages/pandas/core/indexes/base.py", line 5877, in _get_indexer_strict
    self._raise_if_missing(keyarr, indexer, axis_name)
  File "/usr/local/lib/python3.8/dist-packages/pandas/core/indexes/base.py", line 5941, in _raise_if_missing
    raise KeyError(f"{not_found} not in index")
KeyError: "['O_CUSTKEY', 'O_ORDERDATE', 'O_TOTALPRICE', 'L_QUANTITY'] not in index"
Score.fail_execute
python /dbknitter/dbknitter/grader.py /platform/source/s01/v4_enggpt/m00022211/m00022211_q19_t0.py /platform/output/s01/v4_enggpt/m00022211/m00022211_q19_t0.csv /platform/expected/s01/q19.csv
>>> Expected ((1, 1))
      REVENUE
0  168597.286

>>> Actual ((42, 2))
    PartKey     Revenue
0        55  15748.7745
1       424      0.0000
2       433      0.0000
3      1092      0.0000
4      1318   6730.5912
5      2041      0.0000
6      2994      0.0000
7      3140      0.0000
8      3312  19627.2565
9      3746  13808.3238
10     3762  67080.1552
11     3869      0.0000
12     3987  68415.6564
13     5221      0.0000
14     5258      0.0000
15     5287   7010.6064
16     5657      0.0000
17     6182      0.0000
18     6285  28888.5400
19     6421      0.0000
20     6449      0.0000
21     7088      0.0000
22     7594      0.0000
23     7952      0.0000
24    10423      0.0000
25    10857      0.0000
26    12087      0.0000
27    12405  39100.4320
28    13106  18343.8000
29    14013      0.0000
30    15316      0.0000
31    15576   4385.2158
32    16587      0.0000
33    16983  10715.8872
34    17036  17926.4943
35    17115      0.0000
36    17233  41339.2662
37    17318      0.0000
38    18282  51792.0820
39    19146  40688.3480
40    19450  16570.3450
41    19916      0.0000

Score.fail_wrong_output
python /dbknitter/dbknitter/grader.py /platform/source/s01/v4_enggpt/m00022211/m00022211_q1_t0.py /platform/output/s01/v4_enggpt/m00022211/m00022211_q1_t0.csv /platform/expected/s01/q1.csv
>>> Expected ((4, 10))
   AVG_DISC     AVG_PRICE    AVG_QTY  COUNT_ORDER L_LINESTATUS L_RETURNFLAG  SUM_BASE_PRICE    SUM_CHARGE  SUM_DISC_PRICE  SUM_QTY
0  0.050145  36002.123829  25.537587       147790            F            A    5.320754e+09  5.256751e+09    5.054096e+09  3774200
1  0.049394  35521.326916  25.300664         3765            F            N    1.337378e+08  1.322863e+08    1.271324e+08    95257
2  0.050096  36000.924688  25.545538       292000            O            N    1.051227e+10  1.038558e+10    9.986238e+09  7459297
3  0.049989  35994.029214  25.525944       148301            F            R    5.337951e+09  5.274406e+09    5.071819e+09  3785523

>>> Actual ((0, 10))
Empty DataFrame
Columns: [AVG_DISC, AVG_PRICE, AVG_QTY, COUNT_ORDER, LINE_STATUS, RETURN_FLAG, SUM_BASE_PRICE, SUM_CHARGE, SUM_DISC_PRICE, SUM_QTY]
Index: []

Score.fail_wrong_output
python /dbknitter/dbknitter/grader.py /platform/source/s01/v4_enggpt/m00022211/m00022211_q20_t0.py /platform/output/s01/v4_enggpt/m00022211/m00022211_q20_t0.csv /platform/expected/s01/q20.csv
Traceback (most recent call last):
  File "/platform/source/s01/v4_enggpt/m00022211/m00022211_q20_t0.py", line 50, in <module>
    supplier_df = redis_query('supplier')
  File "/platform/source/s01/v4_enggpt/m00022211/m00022211_q20_t0.py", line 34, in redis_query
    return pd.read_json(redis_connection.get(table_name))
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/json/_json.py", line 760, in read_json
    json_reader = JsonReader(
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/json/_json.py", line 861, in __init__
    data = self._get_data_from_filepath(filepath_or_buffer)
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/json/_json.py", line 901, in _get_data_from_filepath
    self.handles = get_handle(
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/common.py", line 707, in get_handle
    if _is_binary_mode(path_or_buf, mode) and "b" not in mode:
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/common.py", line 1167, in _is_binary_mode
    return isinstance(handle, _get_binary_io_classes()) or "b" in getattr(
TypeError: argument of type 'method' is not iterable
Score.fail_execute
python /dbknitter/dbknitter/grader.py /platform/source/s01/v4_enggpt/m00022211/m00022211_q21_t0.py /platform/output/s01/v4_enggpt/m00022211/m00022211_q21_t0.csv /platform/expected/s01/q21.csv
Traceback (most recent call last):
  File "/platform/source/s01/v4_enggpt/m00022211/m00022211_q21_t0.py", line 20, in <module>
    suppliers_df = pd.DataFrame(json.loads(supplier_json_data))
  File "/usr/lib/python3.8/json/__init__.py", line 341, in loads
    raise TypeError(f'the JSON object must be str, bytes or bytearray, '
TypeError: the JSON object must be str, bytes or bytearray, not DataFrame
Score.fail_execute
python /dbknitter/dbknitter/grader.py /platform/source/s01/v4_enggpt/m00022211/m00022211_q22_t0.py /platform/output/s01/v4_enggpt/m00022211/m00022211_q22_t0.csv /platform/expected/s01/q22.csv
Traceback (most recent call last):
  File "/platform/source/s01/v4_enggpt/m00022211/m00022211_q22_t0.py", line 55, in <module>
    process_data()
  File "/platform/source/s01/v4_enggpt/m00022211/m00022211_q22_t0.py", line 24, in process_data
    customer_df = get_redis_customers()
  File "/platform/source/s01/v4_enggpt/m00022211/m00022211_q22_t0.py", line 17, in get_redis_customers
    customer_df = pd.read_json(dredis.get('customer'))
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/json/_json.py", line 760, in read_json
    json_reader = JsonReader(
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/json/_json.py", line 861, in __init__
    data = self._get_data_from_filepath(filepath_or_buffer)
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/json/_json.py", line 901, in _get_data_from_filepath
    self.handles = get_handle(
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/common.py", line 707, in get_handle
    if _is_binary_mode(path_or_buf, mode) and "b" not in mode:
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/common.py", line 1167, in _is_binary_mode
    return isinstance(handle, _get_binary_io_classes()) or "b" in getattr(
TypeError: argument of type 'method' is not iterable
Score.fail_execute
python /dbknitter/dbknitter/grader.py /platform/source/s01/v4_enggpt/m00022211/m00022211_q2_t0.py /platform/output/s01/v4_enggpt/m00022211/m00022211_q2_t0.csv /platform/expected/s01/q2.csv
/platform/source/s01/v4_enggpt/m00022211/m00022211_q2_t0.py:34: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  mysql_df = pd.read_sql(mysql_query, mysql_conn)
Traceback (most recent call last):
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/sql.py", line 2202, in execute
    cur.execute(sql, *args)
  File "/usr/local/lib/python3.8/dist-packages/pymysql/cursors.py", line 153, in execute
    result = self._query(query)
  File "/usr/local/lib/python3.8/dist-packages/pymysql/cursors.py", line 322, in _query
    conn.query(q)
  File "/usr/local/lib/python3.8/dist-packages/pymysql/connections.py", line 558, in query
    self._affected_rows = self._read_query_result(unbuffered=unbuffered)
  File "/usr/local/lib/python3.8/dist-packages/pymysql/connections.py", line 822, in _read_query_result
    result.read()
  File "/usr/local/lib/python3.8/dist-packages/pymysql/connections.py", line 1200, in read
    first_packet = self.connection._read_packet()
  File "/usr/local/lib/python3.8/dist-packages/pymysql/connections.py", line 772, in _read_packet
    packet.raise_for_error()
  File "/usr/local/lib/python3.8/dist-packages/pymysql/protocol.py", line 221, in raise_for_error
    err.raise_mysql_exception(self._data)
  File "/usr/local/lib/python3.8/dist-packages/pymysql/err.py", line 143, in raise_mysql_exception
    raise errorclass(errno, errval)
pymysql.err.ProgrammingError: (1146, "Table 'tpch.partsupp' doesn't exist")

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/platform/source/s01/v4_enggpt/m00022211/m00022211_q2_t0.py", line 34, in <module>
    mysql_df = pd.read_sql(mysql_query, mysql_conn)
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/sql.py", line 635, in read_sql
    return pandas_sql.read_query(
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/sql.py", line 2266, in read_query
    cursor = self.execute(sql, params)
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/sql.py", line 2214, in execute
    raise ex from exc
pandas.errors.DatabaseError: Execution failed on sql '
SELECT 
  p.P_PARTKEY, p.P_NAME, p.P_MFGR, p.P_SIZE, n.N_NATIONKEY, n.N_NAME,
  r.R_NAME, p.P_TYPE, ps.PS_SUPPLYCOST, s.S_ACCTBAL, s.S_NAME,
  s.S_ADDRESS, s.S_PHONE, s.S_COMMENT
FROM
  part p
JOIN
  partsupp ps ON p.P_PARTKEY = ps.PS_PARTKEY
JOIN
  supplier s ON ps.PS_SUPPKEY = s.S_SUPPKEY
JOIN
  nation n ON s.S_NATIONKEY = n.N_NATIONKEY
JOIN
  region r ON n.N_REGIONKEY = r.R_REGIONKEY
WHERE
  p.P_SIZE = 15 AND p.P_TYPE LIKE '%BRASS'
  AND r.R_NAME = 'EUROPE';
': (1146, "Table 'tpch.partsupp' doesn't exist")
Score.fail_execute
python /dbknitter/dbknitter/grader.py /platform/source/s01/v4_enggpt/m00022211/m00022211_q3_t0.py /platform/output/s01/v4_enggpt/m00022211/m00022211_q3_t0.csv /platform/expected/s01/q3.csv
Traceback (most recent call last):
  File "/platform/source/s01/v4_enggpt/m00022211/m00022211_q3_t0.py", line 27, in <module>
    customer_df = pd.read_json(customer_data, orient='records')
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/json/_json.py", line 760, in read_json
    json_reader = JsonReader(
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/json/_json.py", line 861, in __init__
    data = self._get_data_from_filepath(filepath_or_buffer)
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/json/_json.py", line 901, in _get_data_from_filepath
    self.handles = get_handle(
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/common.py", line 707, in get_handle
    if _is_binary_mode(path_or_buf, mode) and "b" not in mode:
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/common.py", line 1167, in _is_binary_mode
    return isinstance(handle, _get_binary_io_classes()) or "b" in getattr(
TypeError: argument of type 'method' is not iterable
Score.fail_execute
python /dbknitter/dbknitter/grader.py /platform/source/s01/v4_enggpt/m00022211/m00022211_q4_t0.py /platform/output/s01/v4_enggpt/m00022211/m00022211_q4_t0.csv /platform/expected/s01/q4.csv
>>> Expected ((5, 2))
   ORDER_COUNT  O_ORDERPRIORITY
0          999         1-URGENT
1          997           2-HIGH
2         1031         3-MEDIUM
3          989  4-NOT SPECIFIED
4         1077            5-LOW

>>> Actual ((0, 2))
Empty DataFrame
Columns: [count, order_priority]
Index: []

Score.fail_wrong_output
python /dbknitter/dbknitter/grader.py /platform/source/s01/v4_enggpt/m00022211/m00022211_q5_t0.py /platform/output/s01/v4_enggpt/m00022211/m00022211_q5_t0.csv /platform/expected/s01/q5.csv
Traceback (most recent call last):
  File "/platform/source/s01/v4_enggpt/m00022211/m00022211_q5_t0.py", line 33, in <module>
    cursor.execute(mysql_query)
  File "/usr/local/lib/python3.8/dist-packages/pymysql/cursors.py", line 153, in execute
    result = self._query(query)
  File "/usr/local/lib/python3.8/dist-packages/pymysql/cursors.py", line 322, in _query
    conn.query(q)
  File "/usr/local/lib/python3.8/dist-packages/pymysql/connections.py", line 558, in query
    self._affected_rows = self._read_query_result(unbuffered=unbuffered)
  File "/usr/local/lib/python3.8/dist-packages/pymysql/connections.py", line 822, in _read_query_result
    result.read()
  File "/usr/local/lib/python3.8/dist-packages/pymysql/connections.py", line 1200, in read
    first_packet = self.connection._read_packet()
  File "/usr/local/lib/python3.8/dist-packages/pymysql/connections.py", line 772, in _read_packet
    packet.raise_for_error()
  File "/usr/local/lib/python3.8/dist-packages/pymysql/protocol.py", line 221, in raise_for_error
    err.raise_mysql_exception(self._data)
  File "/usr/local/lib/python3.8/dist-packages/pymysql/err.py", line 143, in raise_mysql_exception
    raise errorclass(errno, errval)
pymysql.err.OperationalError: (1054, "Unknown column 'l.L_EXTENDEDPRICE' in 'field list'")
Score.fail_execute
python /dbknitter/dbknitter/grader.py /platform/source/s01/v4_enggpt/m00022211/m00022211_q6_t0.py /platform/output/s01/v4_enggpt/m00022211/m00022211_q6_t0.csv /platform/expected/s01/q6.csv
The total revenue has been written to query_output.csv
>>> Expected ((1, 1))
        REVENUE
0  7.115407e+06

>>> Actual ((1, 1))
   Total Revenue
0   1.845191e+08

Head row similarity: 0.4375
Score.fail_wrong_output
python /dbknitter/dbknitter/grader.py /platform/source/s01/v4_enggpt/m00022211/m00022211_q7_t0.py /platform/output/s01/v4_enggpt/m00022211/m00022211_q7_t0.csv /platform/expected/s01/q7.csv
Traceback (most recent call last):
  File "/platform/source/s01/v4_enggpt/m00022211/m00022211_q7_t0.py", line 35, in <module>
    supplier_df = pd.read_json(redis_client.get('supplier'))
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/json/_json.py", line 760, in read_json
    json_reader = JsonReader(
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/json/_json.py", line 861, in __init__
    data = self._get_data_from_filepath(filepath_or_buffer)
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/json/_json.py", line 901, in _get_data_from_filepath
    self.handles = get_handle(
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/common.py", line 707, in get_handle
    if _is_binary_mode(path_or_buf, mode) and "b" not in mode:
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/common.py", line 1167, in _is_binary_mode
    return isinstance(handle, _get_binary_io_classes()) or "b" in getattr(
TypeError: argument of type 'method' is not iterable
Score.fail_execute
python /dbknitter/dbknitter/grader.py /platform/source/s01/v4_enggpt/m00022211/m00022211_q8_t0.py /platform/output/s01/v4_enggpt/m00022211/m00022211_q8_t0.csv /platform/expected/s01/q8.csv
Traceback (most recent call last):
  File "/platform/source/s01/v4_enggpt/m00022211/m00022211_q8_t0.py", line 41, in <module>
    supplier_df = pd.read_json(supplier_data)
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/json/_json.py", line 760, in read_json
    json_reader = JsonReader(
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/json/_json.py", line 861, in __init__
    data = self._get_data_from_filepath(filepath_or_buffer)
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/json/_json.py", line 901, in _get_data_from_filepath
    self.handles = get_handle(
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/common.py", line 707, in get_handle
    if _is_binary_mode(path_or_buf, mode) and "b" not in mode:
  File "/usr/local/lib/python3.8/dist-packages/pandas/io/common.py", line 1167, in _is_binary_mode
    return isinstance(handle, _get_binary_io_classes()) or "b" in getattr(
TypeError: argument of type 'method' is not iterable
Score.fail_execute
python /dbknitter/dbknitter/grader.py /platform/source/s01/v4_enggpt/m00022211/m00022211_q9_t0.py /platform/output/s01/v4_enggpt/m00022211/m00022211_q9_t0.csv /platform/expected/s01/q9.csv
/platform/source/s01/v4_enggpt/m00022211/m00022211_q9_t0.py:17: UserWarning: pandas only supports SQLAlchemy connectable (engine/connection) or database string URI or sqlite3 DBAPI2 connection. Other DBAPI2 objects are not tested. Please consider using SQLAlchemy.
  parts_df = pd.read_sql(part_query, mysql_conn)
Score.fail_timeout
